test/integration/oidc/oidc_test.go

```diff
@@ -0,0 +1,435 @@
+		Subjects: []rbacV1.Subject{
+			{
+				APIGroup: rbac.GroupName,
+				Kind:     rbacV1.UserKind,
+				Name:     defaultOIDCUsernamePrefix + defaultOIDCClaimedUsername,
+			},
+		},
+		RoleRef: rbacV1.RoleRef{
+			APIGroup: rbac.GroupName,
+			Kind:     "Role",
+			Name:     defaultRBACRoleName,
+		},
+	}
+)
+
+func TestIDTokenOk(t *testing.T) {
+	oidcServer, apiServer := configureTestInfrastructure(t)
+
+	idTokenLifetime := time.Second * 1200
+	oidcServer.TokenHandler().EXPECT().Token().Times(1).DoAndReturn(oidcTest.DefaultTokenHandlerBehaviour(
+		t,
+		rsaPrivateKeyFilePath,
+		oidcServer.URL(),
+		defaultOIDCClientID,
+		defaultOIDCClaimedUsername,
+		defaultStubAccessToken,
+		defaultStubRefreshToken,
+		time.Now().Add(idTokenLifetime).Unix(),
+	))
+	idToken, stubRefreshToken := fetchOIDCCredentials(t, oidcServer.TokenURL(), caFilePath)
+
+	configureClientConfigForOIDC(apiServer.ClientConfig, idToken, stubRefreshToken, oidcServer.URL())
+	expectedOkClient := kubernetes.NewForConfigOrDie(apiServer.ClientConfig)
+	_, err := expectedOkClient.CoreV1().Pods(defaultNamespace).List(context.Background(), metaV1.ListOptions{})
+	assert.NoError(t, err)
+}
+
+func TestIDTokenExpired(t *testing.T) {
+	oidcServer, apiServer := configureTestInfrastructure(t)
+
+	expiredIDToken, stubRefreshToken := fetchExpiredToken(t, oidcServer)
+	configureClientConfigForOIDC(apiServer.ClientConfig, expiredIDToken, stubRefreshToken, oidcServer.URL())
+	expiredClient := kubernetes.NewForConfigOrDie(apiServer.ClientConfig)
+
+	// expiredClient(i.e. client with expired token) will try to get new not expired ID token,
+	// let's make OIDC Server to return expired token again
+	configureOIDCServerToReturnExpiredIDToken(t, oidcServer)
+
+	_, err := expiredClient.CoreV1().Pods(defaultNamespace).List(context.Background(), metaV1.ListOptions{})
+	assert.EqualError(t, err, string(metaV1.StatusReasonUnauthorized))
+}
+
+func TestUpdatingRefreshTokenInCaseOfExpiredIDToken(t *testing.T) {
+	var tests = []struct {
+		name                            string
+		configureUpdatingTokenBehaviour func(t *testing.T, oidcServer *oidcTest.TestServer)
+		expectedErr                     error
+	}{
+		{
+			name: "cache returns stale client if refresh token is not updated in config",
+			configureUpdatingTokenBehaviour: func(t *testing.T, oidcServer *oidcTest.TestServer) {
+				t.Helper()
+
+				oidcServer.TokenHandler().EXPECT().Token().Times(1).DoAndReturn(oidcTest.DefaultTokenHandlerBehaviour(
+					t,
+					rsaPrivateKeyFilePath,
+					oidcServer.URL(),
+					defaultOIDCClientID,
+					defaultOIDCClaimedUsername,
+					defaultStubAccessToken,
+					defaultStubRefreshToken,
+					time.Now().Add(time.Second*1200).Unix(),
+				))
+				configureOIDCServerToReturnExpiredRefreshTokenErrorOnTryingToUpdateIDToken(oidcServer)
+			},
+			expectedErr: new(url.Error),
+		},
+		// This test case will be passed when PR
+		// https://github.com/kubernetes/kubernetes/pull/108794
```

Comment:
Merging commented-out content is not a good practice.

Reply:
I don't think we can merge commented code, I would advise adding that piece of code in a review of a PR fixing the issue. You can even open such a PR as a follow-up.